src/config/settings.rs

```diff
@@ -14,6 +15,93 @@ pub enum Setting {
     Map(Vec<String>),
 }
 
+impl Setting {
+    pub fn to_bool(&self) -> bool {
+        if !matches!(self, Setting::Bool(_)) {
+            warn!("setting is not a boolean");
+        }
+
+        match self {
+            Setting::Bool(value) => *value,
+            Setting::SInt(value) => *value != 0,
+            Setting::UInt(value) => *value != 0,
+            Setting::String(value) => is_bool_value(value),
+            Setting::Map(values) => !values.is_empty(),
+        }
+    }
+
+    pub fn to_sint(&self) -> isize {
+        if !matches!(self, Setting::SInt(_)) {
+            warn!("setting is not an signed integer");
+        }
+
+        match self {
+            Setting::SInt(value) => *value,
+            Setting::UInt(value) => *value as isize,
+            Setting::Bool(value) => *value as isize,
+            Setting::String(value) => is_bool_value(value) as isize,
```

Comment:
For `to_sint` and `to_uint`, should the string convert to the actual integer representation or leave it as bool here?

If there was a setting `s:123` stored as a string and we tried converting it to its integer representation then it would return 1 which may be unexpected.

I'm not sure what cases if/when that would ever come up, but something to consider

Reply:
Of course, we shouldn't store integer settings as strings, but.. :P

Reply:
Makes sense to me, to do that.